sdk/nodejs/runtime/dependsOn.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getAllResources, Input, Output } from "../output";
+import { Resource } from "../resource";
+
+/**
+ * Gathers explicit dependent Resources from a list of Resources (possibly Promises and/or Outputs).
+ */
```

Comment:
This should be `@internal` so it doesn't show up in generated docs.

```suggestion
 * @internal
 */
```